### PR TITLE
Revert "Emergency fix: Use stable Docker images (#318)"

### PR DESCRIPTION
**Pull request by mtojek**
This reverts commit 003c019620b2889172c86ba281819c1dea4ec59a - latest emergency fix.

Required changes:
- kibana config: use `xpack.fleet.agents.fleet_server.hosts`
- kibana config: remove deprecated fields
- increase resource limits for agent deployed in Kubernetes
- k8s agent yaml: add RBAC role
- regenerate expected test results for AWS integration (updated GeoDB)
- add missing field for `kubernetes.pod`

### CODE REVIEW

**Comment by mtojek** [comment]
@nchaulet This PR should go back to green once the issue around Kibana/Fleet/Agent is solved.

**Comment by nchaulet** [comment]
@mtojek @blakerouse from what I tested this is working correctly with agent 8.0.0-SNAPSHOT it's not with 7.13 did we miss a backport somewhere or is there a build that failed?

**Comment by blakerouse** [comment]
@nchaulet I have not found a missing backport, I believe it is a snapshot build issue.

**Comment by mtojek** [comment]
It seems that there was a successful build yesterday:

>43 - 7.13.0-b51da292
2021-04-11 00:12

**Comment by mtojek** [comment]
@blakerouse @nchaulet Would you mind sharing the status? Do you need more reference data/logs?

For clarity: due to this issue we are unable to test integrations with latest snapshots (current using ones published a week ago).

**Comment by ruflin** [comment]
I think it is the `kibana.host` config that causes an issue. I assume it was renamed. Let me check the code: https://github.com/elastic/elastic-package/blob/master/internal/install/static_kibana_config_yml.go#L21

**Comment by mtojek** [comment]
If so, then it's breaking for all parties including test environments and e2e tests.

**Comment by ruflin** [comment]
I think fleet-server.hosts must be used: https://github.com/elastic/kibana/blob/master/x-pack/plugins/fleet/server/index.ts#L58 But the other one is only deprecated 🤔  Let me pull down your code an try it out.

**Comment by mtojek** [comment]
I tweaked the config based on the file you linked here. Let's see.

**Comment by ruflin** [comment]
I added the following and removed the kibana part:

```
xpack.fleet.agents.fleet_server.hosts: ["http://fleet-server:8220"]
```

It should be tested if it also works with the Kibana one.

File: test/packages/aws/data_stream/cloudtrail/_dev/test/pipeline/test-add-user-to-group-json.log-expected.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
                 "ip": "127.0.0.1"
             },
             "event": {
-                "ingested": "2021-03-18T12:21:57.668559300Z",
+                "ingested": "2021-04-19T09:58:42.209230300Z",
```
Comment (ruflin) [inline test/packages/aws/data_stream/cloudtrail/_dev/test/pipeline/test-add-user-to-group-json.log-expected.json:22]: Do you know why this is needed?

**Comment by ruflin** [comment]
I tried with keeping kibana.host in addition but didn't work. Lets continue the discussion around this in the Kibana issue. Hoping this goes green, we should move forward.

@mtojek You mentioned the e2e tests. Do you know where exactly we need to update it?

File: test/packages/aws/data_stream/cloudtrail/_dev/test/pipeline/test-add-user-to-group-json.log-expected.json (hunk 1 above)
Comment (mtojek) [reply test/packages/aws/data_stream/cloudtrail/_dev/test/pipeline/test-add-user-to-group-json.log-expected.json:22]: I regenerated test results as it's simple for us to operate, but it still failed somewhere. Investigating.

**Comment by mtojek** [comment]
> @mtojek You mentioned the e2e tests. Do you know where exactly we need to update it?

I noticed this issue that Manu created: https://github.com/elastic/e2e-testing/issues/1048

**Comment by mtojek** [comment]
Yes, I updated the GeoDB references. Let's wait for the CI.

**Comment by mtojek** [comment]
The Kubernetes service deployer complain about the unhealthy agent's pod. Investigating.

**Comment by mtojek** [comment]
I didn't expect this (OOMKilled):

```
Name:         kind-fleet-agent-clusterscope-bf6fdc5c7-5q87q
Namespace:    kube-system
Priority:     0
Node:         kind-control-plane/172.20.0.2
Start Time:   Mon, 19 Apr 2021 11:59:32 +0000
Labels:       app=kind-fleet-agent-clusterscope
              group=fleet
              pod-template-hash=bf6fdc5c7
Annotations:  <none>
Status:       Running
IP:           10.244.0.5
IPs:
  IP:           10.244.0.5
Controlled By:  ReplicaSet/kind-fleet-agent-clusterscope-bf6fdc5c7
Containers:
  kind-fleet-agent-clusterscope:
    Container ID:   containerd://6a0b8a291fdcb44f88b73d46b61a08bebed2f34af6c13e3d05a39f94f503955d
    Image:          docker.elastic.co/beats/elastic-agent:7.13.0-SNAPSHOT
    Image ID:       docker.elastic.co/beats/elastic-agent@sha256:c41329c125066539715de67a53a314e857c49390189a4f9f17d633599b356f14
    Port:           <none>
    Host Port:      <none>
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       OOMKilled
      Exit Code:    137
      Started:      Mon, 19 Apr 2021 12:07:18 +0000
      Finished:     Mon, 19 Apr 2021 12:07:33 +0000
    Ready:          False
    Restart Count:  6
    Limits:
      memory:  200Mi
    Requests:
      cpu:     100m
      memory:  100Mi
    Startup:   exec [sh -c grep "Agent is starting" -r . --include=elastic-agent-json.log] delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      FLEET_ENROLL:    1
      FLEET_INSECURE:  1
      FLEET_URL:       http://fleet-server:8220
      NODE_NAME:        (v1:spec.nodeName)
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kind-fleet-agent-token-xf4zj (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  kind-fleet-agent-token-xf4zj:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  kind-fleet-agent-token-xf4zj
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  10m                    default-scheduler  Successfully assigned kube-system/kind-fleet-agent-clusterscope-bf6fdc5c7-5q87q to kind-control-plane
  Normal   Pulling    10m                    kubelet            Pulling image "docker.elastic.co/beats/elastic-agent:7.13.0-SNAPSHOT"
  Normal   Pulled     9m55s                  kubelet            Successfully pulled image "docker.elastic.co/beats/elastic-agent:7.13.0-SNAPSHOT" in 11.176921115s
  Warning  Unhealthy  9m32s                  kubelet            Startup probe failed:
  Normal   Created    7m7s (x5 over 9m52s)   kubelet            Created container kind-fleet-agent-clusterscope
  Normal   Pulled     7m7s (x4 over 9m35s)   kubelet            Container image "docker.elastic.co/beats/elastic-agent:7.13.0-SNAPSHOT" already present on machine
  Normal   Started    7m6s (x5 over 9m52s)   kubelet            Started container kind-fleet-agent-clusterscope
  Warning  BackOff    5m1s (x15 over 9m16s)  kubelet            Back-off restarting failed container
```

**Comment by ruflin** [comment]
If you increase the memory, will it go through? Is this the container with fleet-server or without?

**Comment by mtojek** [comment]
> If you increase the memory, will it go through?

It went through.

> Is this the container with fleet-server or without?

Without.

I'm investigating the next level issue. It looks like related to the Kubernetes `pod` data stream.

```
[2021-04-19T13:21:53.343Z] Error: error running package system tests: could not complete test run: failed to validate fields: test case failed: one or more errors found in documents stored in metrics-kubernetes.pod-ep data stream
```

Unfortunately that's the price (technical debt) we're paying for not using snapshots immediately after releasing.

**Comment by ruflin** [comment]
So the failing test is related to the package or to our setup?

**Comment by mtojek** [comment]
Currently it's the test integration:

```
[0] field "kubernetes.pod.ip" is undefined
[1] field "kubernetes.pod.ip" is undefined
[2] field "kubernetes.pod.ip" is undefined
[3] field "kubernetes.pod.ip" is undefined
[4] field "kubernetes.pod.ip" is undefined
[5] field "kubernetes.pod.ip" is undefined
[6] field "kubernetes.pod.ip" is undefined
[7] field "kubernetes.pod.ip" is undefined
[8] field "kubernetes.pod.ip" is undefined
[9] field "kubernetes.pod.ip" is undefined
[10] field "kubernetes.pod.ip" is undefined
[11] field "kubernetes.pod.ip" is undefined
[12] field "kubernetes.pod.ip" is undefined
[13] field "kubernetes.pod.ip" is undefined
```

**Comment by ruflin** [comment]
I wonder how this is related to the "stack" change we made?

**Comment by mtojek** [comment]
I confirmed it with @ChrsMark that there was a field added in Beats in the mean time.

**Review by ruflin — APPROVED** [review]
LGTM

**Pull request merged**